artiq/protocols/pc_rpc.py

```diff
@@ -150,6 +150,9 @@ def __recv(self):
             if not more:
                 break
             buf += more.decode()
+        if not buf:
+            self.__socket.close()
+            raise ConnectionError('The RPC server closed the connection')
```

Comment:
That should be a separate PR and done consistently (we have the same code pattern e.g. in sync_struct).

Reply:
Which bit of `sync_struct` do you mean, this part:

https://github.com/m-labs/artiq/blob/e50a6d5aafcc08ecc8ecc30972001cd58c9604a0/artiq/protocols/sync_struct.py#L142-L149

?

Reply:
Yes.

Reply:
So the `sync_struct` treats Publisher disconnection as a normal event, and just quietly shuts down, calling the `disconnect_cb` callback to let the user do whatever's required to handle it. The same approach for `Client` would make sense, so there's no error logged unless the user adds a callback (preserving current behaviour).

Reply:
Can you split this into a separate PR? Keep this one about the server.